### PR TITLE
[Feature] Migrate occluded metric to 3.x.

### PR DESCRIPTION
**Pull request by RangiLyu**
Migrate occluded metic to 3.x.

TODO:
- [x] unit tests

### CODE REVIEW

**Review by BIGWangYuDong — DISMISSED** [review]
LGTM

**Review by hhaAndroid — APPROVED** [review]

**Review by ZwwWayne — DISMISSED** [review]

File: docs/en/user_guides/useful_tools.md [hunk 1]
```diff
@@ -509,3 +509,78 @@ python tools/analysis_tools/confusion_matrix.py ${CONFIG}  ${DETECTION_RESULTS}
...
++-----------+--------+-------------+
+| occluded  | 58.79% | 3263        |
+| separated | 31.94% | 1125        |
++-----------+--------+-------------+
+Evaluation results have been saved to occluded_separated_recall.json.
+```
+
+### Online evaluation
+
+We implement `CocoOccludedSeparatedMetric` which inherited from the `CocoMetic`.
```
Comment (ZwwWayne) [inline docs/en/user_guides/useful_tools.md:565]: inherited -> inherits

File: mmdet/evaluation/metrics/dump_det_results.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+from mmengine.evaluator.metric import _to_cpu
+
+from mmdet.registry import METRICS
+from mmdet.structures.mask import encode_mask_results
+
+
+@METRICS.register_module()
+class DumpDetResults(DumpResults):
+    """Dump model predictions to a pickle file for offline evaluation.
+
```
Comment (ZwwWayne) [inline mmdet/evaluation/metrics/dump_det_results.py:15]: Might indicate what is the difference between it and the DumpResults.

File: mmdet/evaluation/metrics/coco_occluded_metric.py [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+    Args:
+        occluded_ann (str): Path to the occluded coco annotation file.
+        separated_ann (str): Path to the separated coco annotation file.
+        score_thr (float): Score threshold of the detection masks.
+            Defaults to 0.3.
+        iou_thr (float): IoU threshold for the recall calculation.
+            Defaults to 0.75.
+        metric (str | List[str]): Metrics to be evaluated. Valid metrics
+            include 'bbox', 'segm', 'proposal', and 'proposal_fast'.
```
Comment (ZwwWayne) [inline mmdet/evaluation/metrics/coco_occluded_metric.py:54]: should we update the docstring? is proposal supported?

File: mmdet/evaluation/metrics/coco_occluded_metric.py [hunk 4]
```diff
@@ -0,0 +1,211 @@
...
+        return coco_metric_res
+
+    def evaluate_occluded_separated(self, results: List[tuple]) -> dict:
+        """Compute the recall of occluded and separated masks.
+
+        Args:
+            results (list[tuple]): Testing results of the dataset.
+
+        Returns:
+            dict[str, float]: The recall of occluded and separated masks.
```
Comment (ZwwWayne) [inline mmdet/evaluation/metrics/coco_occluded_metric.py:103]: This is inconsistent with that in line 88

File: mmdet/evaluation/metrics/coco_occluded_metric.py (hunk 3 above)
Comment (RangiLyu) [reply mmdet/evaluation/metrics/coco_occluded_metric.py:54]: It supports all metrics that `CocoMetric` has.

**Review by ZwwWayne — APPROVED** [review]

**Pull request merged**